tidy3d/components/data/monitor_data_n2f.py

```diff
@@ -375,35 +474,55 @@ class Near2FarCartesianData(AbstractNear2FarData):
     ...     )
     """
 
-    monitor: Near2FarCartesianMonitor
+    monitor: Near2FarCartesianMonitor = None
 
     Ntheta: Near2FarCartesianDataArray = pd.Field(
-        None,
+        ...,
         title="Ntheta",
         description="Spatial distribution of the theta-component of the N radiation vector.",
     )
     Nphi: Near2FarCartesianDataArray = pd.Field(
-        None,
+        ...,
         title="Nphi",
         description="Spatial distribution of the phi-component of the N radiation vector.",
     )
     Ltheta: Near2FarCartesianDataArray = pd.Field(
-        None,
+        ...,
         title="Ltheta",
         description="Spatial distribution of the theta-component of the L radiation vector.",
     )
     Lphi: Near2FarCartesianDataArray = pd.Field(
-        None,
+        ...,
         title="Lphi",
         description="Spatial distribution of the phi-component of the L radiation vector.",
     )
 
     _contains_monitor_fields = enforce_monitor_fields_present()
 
+    @property
+    def x(self) -> np.ndarray:
+        """X positions."""
+        return self.Ntheta.x.values
+
+    @property
+    def y(self) -> np.ndarray:
+        """Y positions."""
+        return self.Ntheta.y.values
+
+    @property
+    def spherical_coords(self) -> Tuple[np.ndarray, np.ndarray, np.ndarray]:
```

Comment:
One reason I avoided vectorization was the studies we did a while ago for numpy vectorization vs. dask vs. loops, where we found that for small sets of coordinates, vectorization worked well, but for larger grids, loops seemed to be a better compromise in speed and memory. Having said that, those studies were for client-side far field projections, while here it's just converting between coordinates and combining the arrays, so maybe vectorization will be fine here.

Reply:
hm, yea I worried that maybe these were unvectorized for a reason. is there a way we can test?